sdk/servicebus/azure-servicebus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,422 @@
+# Troubleshoot Azure Service Bus client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Service Bus client library for Python.
+
+## Table of contents
+
+* [General troubleshooting](#general-troubleshooting)
+  * [Enable client logging](#enable-client-logging)
+  * [Common exceptions](#common-exceptions)
+  * [Timeouts](#timeouts)
+* [Troubleshooting authentication issues](#troubleshooting-authentication-issues)
+  * [Authentication errors](#authentication-errors)
+  * [Authorization errors](#authorization-errors)
+  * [Connection string issues](#connection-string-issues)
+* [Troubleshooting connectivity issues](#troubleshooting-connectivity-issues)
+  * [Connection errors](#connection-errors)
+  * [Firewall and proxy issues](#firewall-and-proxy-issues)
+  * [Service busy errors](#service-busy-errors)
+* [Troubleshooting message handling issues](#troubleshooting-message-handling-issues)
+  * [Message lock issues](#message-lock-issues)
+  * [Message size issues](#message-size-issues)
+  * [Message settlement issues](#message-settlement-issues)
+  * [Dead letter queue issues](#dead-letter-queue-issues)
+* [Troubleshooting session handling issues](#troubleshooting-session-handling-issues)
+  * [Session lock issues](#session-lock-issues)
+  * [Session cannot be locked](#session-cannot-be-locked)
+* [Troubleshooting quota and capacity issues](#troubleshooting-quota-and-capacity-issues)
+  * [Quota exceeded errors](#quota-exceeded-errors)
+  * [Entity not found errors](#entity-not-found-errors)
+* [Frequently asked questions](#frequently-asked-questions)
+* [Get additional help](#get-additional-help)
+
+## General troubleshooting
+
+Azure Service Bus client library will raise exceptions defined in [Azure Core](https://aka.ms/azsdk/python/core/docs#module-azure.core.exceptions) and [azure.servicebus.exceptions](https://docs.microsoft.com/python/api/azure-servicebus/azure.servicebus.exceptions).
+
+### Enable client logging
+
+This library uses the standard [logging](https://docs.python.org/3/library/logging.html) library for logging.
+
+Basic information about HTTP sessions (URLs, headers, etc.) is logged at `INFO` level.
+
+Detailed `DEBUG` level logging, including request/response bodies and **unredacted** headers, can be enabled on the client or per-operation with the `logging_enable` keyword argument.
+
+To enable client logging and AMQP frame level trace:
+
+```python
+import logging
+import sys
+
+handler = logging.StreamHandler(stream=sys.stdout)
+log_fmt = logging.Formatter(fmt="%(asctime)s | %(threadName)s | %(levelname)s | %(name)s | %(message)s")
+handler.setFormatter(log_fmt)
+logger = logging.getLogger('azure.servicebus')
+logger.setLevel(logging.DEBUG)
+logger.addHandler(handler)
+
+# Enable AMQP frame level trace
+from azure.servicebus import ServiceBusClient
+
+client = ServiceBusClient(connection_string, logging_enable=True)
```

Comment:
This should be replaced with Azure Identity credential auth, not connection string.